docs/RFCS/schema_gossip.md

```diff
@@ -0,0 +1,34 @@
+- Feature Name: schema_distribution
+- Start Date: 2015-07-20
+- RFC PR:
+- Cockroach Issue:
+
+# Summary
+
+This RFC suggests implementing eventually-consistent replication of the SQL schema to all Cockroach nodes in a cluster using gossip.
+
+# Motivation
+
+In order to support performant SQL queries, each gateway node must be able to address the data requested by the query. Today this requires the node to read `TableDescriptor`s from the KV map, which we believe (though we haven't measured) will cause a substantial performance hit which we can eliminate by actively gossiping the necessary metadata.
+
+# Detailed design
+
+Distribution of the schema metadata will be modeled after the current configuration distribution mechanism. Whenever a new `TableDescriptor` is written to the KV map, the node holding the leader lease for that key's range will add the `TableDescriptor` to gossip under a key derived from the `TableDescriptor`'s `Name` with a TTL TBD.
+
+On the read side, `sqlserver.Server.getTableDesc`'s behaviour will change such that it reads from gossip rather than the KV store. This will require plumbing a referece to the `Gossip` instance down to the `sqlserver.Server` instance.
+
+# Drawbacks
+
+This is slightly more complicated than the current implementation. Because the schema is eventually-consistent, how do we know when migrations are done? We'll have to count on the TTL, which feels a little dirty.
```

Comment:
This isn't just migrations. The namespace and table descriptors will probably be holding permissions (and possibly other settings, though I can't think of any right now). These changes will potentially be more frequent than schema changes, and will probably need to be applied faster.